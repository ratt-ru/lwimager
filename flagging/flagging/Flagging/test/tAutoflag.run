#!/bin/sh
#-----------------------------------------------------------------------------
# Usage: tAutoflag.run
#-----------------------------------------------------------------------------
# This script executes the tAutoflag.g test script to check if all the
# flagging modules are working
#
# $Id: tAutoflag.run,v 19.4 2004/08/24 21:59:16 cvsmgr Exp $
#-----------------------------------------------------------------------------

  # This script has problems, because it uses glish. So exit immediately.
  echo "UNTESTED: tAutoflag.run (AIPSPATH not defined)"
  exit 3

  if [ "$AIPSPATH" = "" ]
  then
     echo "UNTESTED: tAutoflag.run (AIPSPATH not defined)"
     exit 3
  fi

  # establish location of test MS
  MS='WSRT-test.MS2'
  AIPSDEMO=`echo $AIPSPATH | awk '{printf("%s/data/demo/autoflag",$1)}'`
  MSPATH="$AIPSDEMO/$MS"
  if [ ! -d "$MSPATH" -o ! -w "$MSPATH" ]; then
    MSPATH="$HOME/aips++/data/demo/autoflag/$MS"
    if [ ! -d "$MSPATH" -o ! -w "$MSPATH" ]; then
      echo "UNTESTED: tAutoflag.run (found neither $AIPSDEMO/$MS nor $MSPATH)"
      exit 3
    fi
  fi
  # set symlink
  LINK='tAutoflag_tmp.MS2'
  rm -f $LINK 2>/dev/null
  ln -s $MSPATH $LINK
  if [ ! $? ]; then
    echo "UNTESTED: tAutoflag.run (failed to set $LINK symlink)"
    exit 3
  fi
  
  # run the glish script
  echo ">>>"
  glish -l tAutoflag.g
