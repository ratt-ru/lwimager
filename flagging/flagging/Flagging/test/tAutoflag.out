>>>

 glishrc: mode= Operational (Dwingeloo)

 Path used by Glish (system.path.include, set in .glishrc):
.
/aips++/jnoordam/wbork/
/aips++/daily/dop07_egcs/libexec
system.path.glishelp: /aips++/jnoordam/wbork/
 
>>>
NORMAL: defaultservers ready
NORMAL: dowait variable is T
NORMAL: Starting server timer
NORMAL: Starting server misc
NORMAL: Starting server quanta
NORMAL: Starting server autoflag
NORMAL: Server started: /dop07_0/oms/aips++/code/trial/implement/Flagging/test/autoflag (AIPS++ version: 1.7 (build #44))
NORMAL: Server started: misc (AIPS++ version: 1.7 (build #44))
NORMAL: Server started: quanta (AIPS++ version: 1.7 (build #44))
NORMAL: attached MS tAutoflag.MS2: 6656 rows, 104 times, 105 baselines
<<<
Added method 1: timemed (TimeMedian)
    thr         = 5
    hw          = 10
    rowthr      = 10
    rowhw       = 6
    norow       = F
    column      = DATA
   *expr        = ABS I
    debug       = F
    fignore     = F
Added method 2: freqmed (FreqMedian)
    thr         = 5
    hw          = 10
    rowthr      = 10
    rowhw       = 6
    norow       = F
    column      = DATA
   *expr        = ABS I
    debug       = F
    fignore     = F
Added method 3: sprej (SpectralRej)
    region      = F
    spwid       = F
    fq          = F
   *chan        = [5 30] 
    ndeg        = 2
    rowthr      = 5
    rowhw       = 6
    column      = DATA
   *expr        = ABS I
    debug       = F
    fignore     = F
Added method 4: uvbin (UVBinner)
   *thr         = 0.04
    minpop      = 0
    nbins       = 50
   *plotchan    = 100
    econoplot   = T
    column      = DATA
   *expr        = ABS I
    fignore     = F
Added method 5: select (Selector)
    spwid       = F
    field       = F
   *fq          = [806 808] 
    chan        = F
    corr        = F
   *ant         = RT6
    baseline    = F
    timerng     = F
    autocorr    = F
    timeslot    = F
    dtime       = 10
    quack       = F
    clip        = F
    flagrange   = F
    unflag      = F
Added method 6: select (Selector)
    spwid       = F
    field       = F
    fq          = F
    chan        = F
    corr        = F
    ant         = F
    baseline    = F
    timerng     = F
    autocorr    = F
    timeslot    = F
    dtime       = 10
    quack       = F
   *clip        = [expr=ABS I, max=10]
    flagrange   = F
    unflag      = F
NORMAL: Starting flagger::run
NORMAL: Existing flags will be reset
NORMAL: TimeMedian: (DATA)ABS(I); thr=5.0 rowthr=10.0  hw=10
NORMAL: Existing flags will be reset
NORMAL: FreqMedian: (DATA)ABS(I); thr=5.0 rowthr=10.0  hw=10
NORMAL: Existing flags will be reset
NORMAL: SpectralRej: (DATA)ABS(I); #4-29; ndeg=2 rowthr=5.0 rowhw=6 
NORMAL: Existing flags will be reset
NORMAL: UVBinner: (DATA)ABS(I); thr=0.04 nbins=50,50 
NORMAL: Existing flags will be reset
NORMAL: Selector: fq=806.00-808.00MHz ant=RT6; flag all 
NORMAL: Existing flags will be reset
NORMAL: Selector#2: flag ABS(I)>10 
NORMAL: Chunk 1:  XX YY, 256 channels, 104 time slots, 105 baselines, 6656 rows
NORMAL: 26 channels will be fitted in this chunk
NORMAL: defaultaipsrc (drc) ready
NORMAL: defaultwidgetserver (dws) ready
NORMAL: 4830 (72.57%) rows have been flagged.
NORMAL: 150310 of 3407872 (4.41%) pixels have been flagged.
NORMAL: TimeMedian: 16034 pixel flags, 1438 row flags
NORMAL: FreqMedian: 25683 pixel flags, 954 row flags
NORMAL: SpectralRej: 0 pixel flags, 4796 row flags
NORMAL: UVBinner: 63722 pixel flags, 0 row flags
NORMAL: Selector: 53040 pixel flags, 0 row flags
NORMAL: Selector#2: 46456 pixel flags, 0 row flags
NORMAL: Flagging complete
NORMAL: 
>>>
NORMAL: Finished flagger::run
      99.63 real       71.49 user        0.97 system
<<<
